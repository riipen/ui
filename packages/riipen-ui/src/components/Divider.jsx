import clsx from "clsx";
import PropTypes from "prop-types";
import React from "react";

import ThemeContext from "../styles/ThemeContext";
import withClasses from "../utils/withClasses";

class Divider extends React.Component {
  static propTypes = {
    /**
     * List of additional classes to apply to this component.
     */
    classes: PropTypes.array,

    /**
     * The variant to use.
     */
    variant: PropTypes.oneOf(["fullWidth", "middle"])
  };

  static defaultProps = {
    classes: [],
    variant: "fullWidth"
  };

  static contextType = ThemeContext;

  render() {
    const { classes, variant } = this.props;

    const theme = this.context;

    const className = clsx(variant, classes);

    return (
      <React.Fragment>
        <div className={className} />
        <style jsx>{`
          div {
            background-color: ${theme.palette.divider};
            height: 1px;
          }

          .middle {
            margin-left: ${theme.spacing(2)}px;
            margin-right: ${theme.spacing(2)}px;
          },
        `}</style>
      </React.Fragment>
    );
  }
}

export default withClasses()(Divider);
