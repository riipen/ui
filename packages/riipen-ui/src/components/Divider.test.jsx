import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import Divider from "./Divider";

describe("<Divider>", () => {
  it("renders without errors", () => {
    let error;

    try {
      mount(<Divider />);
    } catch (e) {
      error = e;
    }

    expect(error).toEqual(undefined);
  });

  it("renders correct snapshot", () => {
    const wrapper = mount(<Divider />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe("default props", () => {
    it("sets correct default props", () => {
      const defaultProps = new Divider().type.defaultProps;

      const wrapper = mount(<Divider />);

      const component = wrapper.find("Divider");
      expect(component.props().variant).toEqual(defaultProps.variant);
    });
  });

  describe("classes prop", () => {
    it("applies all classes to the root node of Divider", () => {
      const classes = ["classOne", "classTwo"];

      const wrapper = mount(<Divider classes={classes} />);

      expect(
        wrapper
          .find("Divider")
          .childAt(0)
          .hasClass(classes[0])
      ).toEqual(true);
      expect(
        wrapper
          .find("Divider")
          .childAt(0)
          .hasClass(classes[1])
      ).toEqual(true);
    });

    it("appends higher order values to classes prop with withClasses decorator", () => {
      const classes = ["riipen", "riipen-divider"];

      const wrapper = mount(<Divider />);

      expect(
        wrapper
          .find("Divider")
          .props()
          .classes.sort()
      ).toEqual(classes.sort());
    });

    it("gives an error when classes are provided as one string", () => {
      const classes = "classOne classTwo";
      const errors = jest.spyOn(console, "error").mockImplementation();

      mount(<Divider classes={classes} />);

      expect(errors).toHaveBeenCalled();
    });
  });

  describe("variant prop", () => {
    it("sets variant class name with a valid variant", () => {
      const variant = "middle";

      const wrapper = mount(<Divider variant={variant} />);

      expect(
        wrapper
          .find("Divider")
          .childAt(0)
          .hasClass(variant)
      ).toEqual(true);
    });

    it("gives an error when given an invalid variant", () => {
      const errors = jest.spyOn(console, "error").mockImplementation();
      const variant = "somethingInvalid";

      mount(<Divider variant={variant} />);

      expect(errors).toHaveBeenCalledTimes(1);
    });
  });
});
