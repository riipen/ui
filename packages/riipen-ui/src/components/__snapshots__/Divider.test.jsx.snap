// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider> renders correct snapshot 1`] = `
<withClasses(Divider)>
  <Divider
    classes={
      Array [
        "riipen",
        "riipen-divider",
      ]
    }
    variant="fullWidth"
  >
    <div
      className="jsx-1596250350 fullWidth riipen riipen-divider"
    />
    <JSXStyle
      dynamic={
        Array [
          "#ddd",
          8,
          8,
        ]
      }
      id="3816656998"
    />
  </Divider>
</withClasses(Divider)>
`;
